src/makeTxn.ts

```diff
@@ -1,1448 +1,798 @@
- * @param receiver - string representation of Algorand address of recipient
- * @param amount - integer amount to send, in microAlgos
- * @param closeRemainderTo - optionally close out remaining account balance to this account, represented as string rep of Algorand address
- * @param note - uint8array of arbitrary data for sender to store
- * @param suggestedParams - a dict holding common-to-all-txns args:
- * fee - integer fee per byte, in microAlgos. for a flat fee, set flatFee to true
- * flatFee - bool optionally set this to true to specify fee as microalgos-per-txn
- *       If true, txn fee may fall below the ALGORAND_MIN_TX_FEE
- * firstValid - integer first protocol round on which this txn is valid
- * lastValid - integer last protocol round on which this txn is valid
- * genesisHash - string specifies hash genesis block of network in use
- * genesisID - string specifies genesis ID of network in use
- * @param rekeyTo - rekeyTo address, optional
+ * Create a new payment transaction
+ *
+ * @param options - Payment transaction parameters
  */
-export function makePaymentTxnWithSuggestedParams(
-  sender: PaymentTxn['sender'],
-  receiver: PaymentTxn['receiver'],
-  amount: PaymentTxn['amount'],
-  closeRemainderTo: PaymentTxn['closeRemainderTo'],
-  note: PaymentTxn['note'],
-  suggestedParams: MustHaveSuggestedParams<PaymentTxn>['suggestedParams'],
-  rekeyTo?: PaymentTxn['rekeyTo']
-) {
-  const o: PaymentTxn = {
+export function makePaymentTxnWithSuggestedParamsFromObject({
+  sender,
+  receiver,
+  amount,
+  closeRemainderTo,
+  suggestedParams,
+  note,
+  lease,
+  rekeyTo,
+}: PaymentTransactionParams & CommonTransactionParams): Transaction {
+  return new Transaction({
+    type: TransactionType.pay,
     sender,
-    receiver,
-    amount,
-    closeRemainderTo,
     note,
-    suggestedParams,
-    type: TransactionType.pay,
+    lease,
     rekeyTo,
-  };
-  return new txnBuilder.Transaction(o);
-}
-
-// helper for above makePaymentTxnWithSuggestedParams, instead accepting an arguments object
-export function makePaymentTxnWithSuggestedParamsFromObject(
-  o: Expand<
-    Pick<
-      RenameProperty<MustHaveSuggestedParams<PaymentTxn>, 'rekeyTo', 'rekeyTo'>,
-      | 'sender'
-      | 'receiver'
-      | 'amount'
-      | 'closeRemainderTo'
-      | 'note'
-      | 'suggestedParams'
-      | 'rekeyTo'
-    >
-  >
-) {
-  return makePaymentTxnWithSuggestedParams(
-    o.sender,
-    o.receiver,
-    o.amount,
-    o.closeRemainderTo,
-    o.note,
-    o.suggestedParams,
-    o.rekeyTo
-  );
+    suggestedParams,
+    paymentParams: {
+      receiver,
+      amount,
+      closeRemainderTo,
+    },
+  });
 }
 
 /**
- * makeKeyRegistrationTxnWithSuggestedParams takes key registration arguments and returns a Transaction object for
- * that key registration operation
+ * Create a new key registration transaction
  *
- * @param sender - string representation of Algorand address of sender
```

Comment:
almost all of the diff in this file is this doc removal but in fact the same params are passed as object props. Is it possible somehow convert this doc to that object properties documentation/description?

Reply:
The docs that were removed were redundant, and in a lot of cases incorrect/outdated. There are better docs on each field in the object param definitions, e.g. https://github.com/algorand/js-algorand-sdk/blob/9d205bd3e7be1c4d97190446b0b746f0d6f2296c/src/types/transactions/base.ts#L480-L483

These should still show up on hover in IDEs (at least for vscode). I'm less happy about how they show up on the generated docs site:
![Screenshot 2024-02-09 at 10 10 55 AM](https://github.com/algorand/js-algorand-sdk/assets/5856867/04385b24-7428-437c-8de4-1d40af8952f2)

You have to click on `PaymentTransactionParams` or `CommonTransactionParams` to see the definitions and their properties, but the docs are still there